adapter/pkg/oasparser/envoyconf/routesWithClusters.go

```diff
@@ -309,19 +391,38 @@ func createRoute(title string, xWso2Basepath string, version string, endpoint mo
 						},
 						Regex: xWso2Basepath,
 					},
-					Substitution: endpoint.Basepath,
+					Substitution: endpointBasepath,
 				},
-				ClusterSpecifier: clusterSpecifier,
 			},
 		}
 	} else {
 		action = &routev3.Route_Route{
 			Route: &routev3.RouteAction{
 				HostRewriteSpecifier: hostRewriteSpecifier,
-				ClusterSpecifier:     clusterSpecifier,
 			},
 		}
 	}
+
+	//TODO: (VirajSalaka) Introduce a separate function
+	if !(prodClusterName == "" || sandClusterName == "") {
```

Comment:
Simplify the boolean logic

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/1525/commits/452e70df3c574001865304cc59dcf25ad30a79e9